### PR TITLE
Issue-36: Add filter to specify which post types can be selected in Newsletter

### LINKED ISSUE #36: Add filter to specify which post types can be selected in Newsletter
### Description

The developer should be able to specify which post types show up in the Post Selector and Multi Post Selector via a filter.

This should be a PHP filter passed to javascript via `wp_localize_script` for consistent developer experience.

### Use Case

Some sites will have custom post types they want to include in their newsletter such as Podcast episodes.

### PR DESCRIPTION
**Pull request by nikkifurls**
### Summary

This pull request adds the functionality to specify which post types show up in the Post Selector and Multi Post Selector via a filter, addressing the need for sites to include custom post types like Podcast episodes in their newsletters. Fixes #36

### Changes

- Implement a filter to allow specification of post types in the Post Selector and Multi Post Selector.

### Test Plan

- Verify that the filter correctly alters the available post types in the Post Selector and Multi Post Selector.
- Ensure that custom post types can be included and work as expected.

### Additional Notes

- This enhancement was requested to improve the flexibility of the Newsletter builder, allowing for a broader range of content types to be included in newsletters.

### CODE REVIEW

**Review by mogmarsh — CHANGES_REQUESTED** [review]
This looks good, but I would like the filtering to happen in PHP, then pass the value to javascript via `wp_localize_script` on the post block edit script. I think this will be a more consistent developer experience, as there are already other php filters, and if we ever need this filtered value in PHP, we'll have it. I should have put this requirement in the original issue. I apologize. I will go add it there now.

Let me know if you have questions.

**Comment by nikkifurls** [comment]
@mogmarsh no worries! I just made that change if you want to 👀 again. TY!

File: blocks/section/edit.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { dispatch, select } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 import { PanelBody, PanelRow } from '@wordpress/components';
 import { useEffect } from '@wordpress/element';
+import { applyFilters } from '@wordpress/hooks';
```
Comment (mogmarsh) [inline blocks/section/edit.tsx]: can remove

**Review by mogmarsh — APPROVED** [review]
there's one line you can delete, otherwise looks good! 👍

File: blocks/section/edit.tsx (hunk 1 above)
Comment (nikkifurls) [reply blocks/section/edit.tsx]: Whoopsie! Thank you!

**Pull request merged**